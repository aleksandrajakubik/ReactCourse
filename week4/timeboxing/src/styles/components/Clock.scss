.Clock {
    color: orangered;

    &__hours {
        color: darkred;
    }
    &__minutes {
        color: green;
    }
    &__seconds {
        color: darkolivegreen;
    }
    &__miliseconds {
        color: purple;
    }
    &__separator {
        color: pink;
    }
}