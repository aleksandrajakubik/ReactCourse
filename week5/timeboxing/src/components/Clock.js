import React from "react";

function isLengthEqual1(stringToCheck) {
    let newString = stringToCheck;
    return newString.length === 1 ? newString = '0' + newString : newString;

}

function hoursValidation(hour) {
    if (hour < 0){
        return String(0);
    } else if (hour > 23) {
        return String(23);
    } else {
        return String(hour);
    }
};


function minutesAndSecondsValidation(time) {
    if (time < 0) {
        return '00'
    } else if (time > 59) {
        return '59'
    } else {
        return String(time);
    }
};

function milisecondsValidation(miliseconds) {
    if (miliseconds < 0) {
        return '000'
    } else if (miliseconds > 999) {
        return '999'
    } else {
        let newMiliseconds = String(miliseconds);
        return newMiliseconds.length === 1 ? newMiliseconds = '00' + newMiliseconds : newMiliseconds.length === 2 ? newMiliseconds = '0' + newMiliseconds : newMiliseconds;
    }
}

function Clock({ className = "", hours = 0, minutes = 2, seconds = 9, miliseconds = 0}) {
    let newHours = isLengthEqual1(hoursValidation(hours));
    let newMinutes = isLengthEqual1(minutesAndSecondsValidation(minutes));
    let newSeconds = isLengthEqual1(minutesAndSecondsValidation(seconds));
    let newMiliseconds = milisecondsValidation(miliseconds);
    // throw new Error("AAAAAAAAAAA!");
    return (
        <h2 className = {"Clock " + className}>Time left: 
        <span className = "Clock__hours">{newHours}</span>
        <span className = "Clock__separator">:</span>
        <span className = "Clock__minutes">{newMinutes}</span>
        <span className = "Clock__separator">:</span>
        <span className = "Clock__seconds">{newSeconds}</span>.
        <span className = "Clock__miliseconds">{newMiliseconds}</span></h2>
    )
}

export default Clock;
