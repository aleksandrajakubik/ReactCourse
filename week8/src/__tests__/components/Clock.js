import React from "react";
import Clock from "../../components/Clock";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

let root = null;
let clockRenderer = null;
describe("<Clock />", () => {
    describe("when given minutes and seconds (DOM)", () => {
        beforeEach(() => {
            root = document.createElement("div");
            ReactDOM.render(
                <Clock minutes={10} seconds={20} />, root
            )
        });

        it("renders h2", () => {
            expect(root.childNodes[0].nodeName).toEqual("H2");
        });

        it("sets a Clock className", () => {
            expect(root.childNodes[0].className).toMatch(/Clock/);
        });

        it("renders time properly", () => {
            expect(root.childNodes[0].textContent).toMatch(/10:20/);
        });
    }); 

    it("sets className to empty string if not given anything else", () => {
        expect(<Clock minutes={10} seconds={20} />).toEqual(<Clock className="" hours={0} miliseconds={0} minutes={10} seconds={20} />)
    });

    describe("when given minutes and seconds (TestRenderer)", () => {
        beforeEach(() => {
            clockRenderer = renderer.create(
                <Clock minutes={10} seconds={20} />
            );  
        });

        it("renders properly", () => {
            expect(clockRenderer.toJSON()).toMatchSnapshot();

        });

        it("renders h2", () => {
            expect(clockRenderer.toJSON().type).toEqual("h2");
        });

        it("sets a Clock className", () => {
            expect(clockRenderer.toJSON().props).toMatchObject({"className": expect.stringMatching(/Clock/)});
        });
    }); 

})