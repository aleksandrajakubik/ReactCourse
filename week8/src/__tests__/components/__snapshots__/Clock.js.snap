// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> when given minutes and seconds (TestRenderer) renders properly 1`] = `
<h2
  className="Clock "
>
  Time left:
  <span
    className="Clock__hours"
  >
    00
  </span>
  <span
    className="Clock__separator"
  >
    :
  </span>
  <span
    className="Clock__minutes"
  >
    10
  </span>
  <span
    className="Clock__separator"
  >
    :
  </span>
  <span
    className="Clock__seconds"
  >
    20
  </span>
  .
  <span
    className="Clock__miliseconds"
  >
    000
  </span>
</h2>
`;
